src/world/types.rs

```diff
@@ -0,0 +1,72 @@
+use mlua::prelude::{LuaError, LuaFunction, LuaTable, LuaValue};
+use mlua::{FromLua, Lua, Result, ToLua};
+
+pub struct Controls<'lua> {
+    pub gun: bool,
+    pub engine: LuaTable<'lua>,
+}
+
+impl<'lua> FromLua<'lua> for Controls<'lua> {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> Result<Self> {
+        match value {
+            LuaValue::Table(table) => Ok(Controls {
+                gun: table.get("gun")?,
+                engine: table.get("engine")?,
+            }),
+            _ => Err(LuaError::FromLuaConversionError {
+                from: value.type_name(),
+                to: "Controls",
+                message: Some("expected table".to_string()),
+            }),
+        }
+    }
+}
+
+pub struct ModuleInputs<'lua> {
+    pub dt: f64,
+    //pub body: LuaUserData, // TODO: Figure out how to use LuaUserData
```

Comment:
I wasn't able to get user data to convert from Lua to Rust this week. It's not needed in missile_launcher, so we can come back to this later.